scripts/tile_benchmark/benchmark.py

```diff
@@ -0,0 +1,365 @@
+
+
+def get_data(host, image_id, requested_levels, requested_num_tiles):
+    """get request information from each server an request tiles"""
+    num_levels = -1
+    data = {}
+    # get data about server, host url and params
+    if host == "atlas":
+
+        data = atlas_init()
+    elif host == "adrc":
+
+        data = adrc_init()
+    else:
+        data = girder_init(host, image_id)
+    num_levels = data[NUM_LEVELS]
+    # if successful begin to grab tiles from each level
+    if num_levels != -1:
+        if requested_levels > num_levels:
+            requested_levels = num_levels
+        # get elapsed time info numpy array include
+        elapsed_time_arr, count = \
+            iterate_over_tiles(
+                host,
+                data,
+                requested_levels,
+                requested_num_tiles)
+
+        # create a new array of correct size and plot
+
+        return elapsed_time_arr
+
+
+def iterate_over_tiles(host, data, requested_levels, requested_num_tiles):
+    """iterate from level 0 to higher levels requesting tiles"""
+    total_tiles = 0
+    for z in range(requested_levels):
+        total_tiles += 2 ** (z * 2)
+
+    count = 0
+    elapsed_time_arr = np.zeros([total_tiles], dtype=float)
+    # collect from all levels
+    for z in range(requested_levels):
+        print"started level %d" % (z)
+
+        for x in range(2 ** z):
+            for y in range(2 ** z):
+
+                if host == "atlas":
+
+                    request_time = atlas_request(data, z, x, y)
+                elif host == "adrc":
+
+                    request_time = adrc_request(data, z, x, y)
+                else:
+
+                    request_time = girder_request(data, z, x, y)
+                if request_time != -1:
+                    elapsed_time_arr[count] = request_time
+                    count += 1
+                    if count == requested_num_tiles:
+                        elapsed_arr_final = np.zeros([count], dtype=float)
+                        np.copyto(elapsed_arr_final, elapsed_time_arr[:count])
+                        return elapsed_arr_final, count
+    print" Could not query %d tiles with only %d levels " \
+         "for host %s could only query %d tiles" % (
+             requested_num_tiles, requested_levels, host, count)
+    elapsed_arr_final = np.zeros([count], dtype=float)
+    np.copyto(elapsed_arr_final, elapsed_time_arr[:count])
+    return elapsed_arr_final, count
+
+
+def adrc_init():
+    data = {}
+    data[
+        URL] = "http://digitalslidearchive.emory.edu/fcgi-bin/iipsrv" \
+               "Openslide.fcgi?DeepZoom=/GLOBAL_SCRATCH/PREUSS_LAB/BATCH1/" \
+               "19775.svs_files/%d/%d_%d.jpg"
+    data[NUM_LEVELS] = 10
+    return data
+
+
+def atlas_init():
+    data = {}
+    num_level = 9
+    image_id = '500c3e674834a3119800000c'
+    database_id = '5074589002e31023d4292d83'
+    data[URL] = "https://slide-atlas.org/tile?img={img_id}&db={db_id}" \
```

Comment:
Girder like to use the format operator `%` instead of the `.format` function, just because the operator handles unicode more gracefully. It doesn't directly matter here, but we should be consistant.

You should be able to do:
```
my_argument = 'bar'
'foo %(my_arg)s baz' % dict(my_arg=my_argument)
```